### ISSUE TITLE
`roc check` hangs if an interface in a platform tries to import a builtin module

### ISSUE THREAD

**Issue by rtfeldman** [roc-lang/roc#3561]
If I have `imports [Str]` in an interface module that's imported by a platform module, `roc check` hangs.

It doesn't hang if that interface module is (directly) imported by an application module, however.

**Comment by lukewilliamboswell** [comment]
Another repro.

Importing a module with the same name as a builtin hangs. 

```
app [main] {
    pf: platform "https://github.com/roc-lang/basic-cli/releases/download/0.11.0/SY4WWMhWQ9NvQgvIthcv15AUeA7rAIJHAHgiaSHGhdY.tar.br",
}

import pf.Task
import pf.List

main = Task.ok {}
```

**Comment by agu-z** [comment]
Fixed by: #6863

**Comment by smores56** [comment]
The fix has been merged, closing.
